index.js

```diff
@@ -174,20 +173,26 @@ function parseStyles(
 ) {
   var statements = parseStatements(result, styles)
 
-  return Promise.resolve(statements).then(promiseEach(function(stmt) {
-    stmt.media = joinMedia(media, stmt.media || [])
+  return Promise.resolve(statements).then(function(stmts) {
+    return stmts.reduce(function(promise, stmt) {
+      return promise.then(function() {
```

Comment:
LGTM, but perhaps we should have a comment here to explain that this cleverness works on the array in series.